litestar/middleware/compression/facade.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, ClassVar, Protocol
+
+if TYPE_CHECKING:
+    from io import BytesIO
+
+    from litestar.config.compression import CompressionConfig
+    from litestar.enums import CompressionEncoding
+
+
+class CompressionFacade(Protocol):
+    """A unified facade offering a uniform interface for different compression libraries."""
+
+    encoding: ClassVar[str]
```

Comment:
This could be typed as `ClassVar[CompressionEncoding]`, but that would mean the user would have to subclass `CompressionEncoding` in order to extend it before giving the value here. I didn't feel like it had enough benefit to do that.